### PR TITLE
#10539 geo.rst doc edit

### PR DESCRIPTION
**Pull request by pillemer**
Editing geo.rst for issue #10539.
First time contributor so let me know if anything is wrong or off. Eager to learn and to help.

All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [ ] issues: fixes #xxxx
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

File: sphinx/source/docs/user_guide/geo.rst [hunk 1]
```diff
@@ -1,59 +1,54 @@
...
-Mapping Geo Data
+Mapping geo data
 ================
 
-Bokeh has started adding support for working with Geographical data. There are
-a number of powerful features already available, but we still have more to add.
-Please tell us your use cases through `Discourse`_ or on `GitHub`_ so that we
-can continue to extend these features to meet your needs.
+Bokeh has started adding support for working with geographical data. There are
+a number of powerful features already available. Tell us your use cases through `Discourse`_ or on `GitHub`_ so that we can continue to extend these features to meet your needs.
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/geo.rst:7]: As long as we are editing this file, I think this old intro para can just be deleted. We are past the point of "starting" and so this is just extra distraction.

**Comment by bryevdv** [comment]
HI @pillemer thanks for the PR! There is currently one codebase/linter issue to fix:
```
E         File contains trailing whitespace: sphinx/source/docs/user_guide/geo.rst, line 16.
```
basically, we check to make sure there is not any trailing whitespace on any lines, so the fix is just to delete the whitespace. In case you are not yet familiar: you can just continue to commit new changes to your existing branch and then push to GitHub, and this PR will update to reflect the new changes automatically. 

I also left a small comment about the intro paragraph I think it can be just deleted. Hopefully @tcmetzger will be able to give a quick once-over as well. Thanks again!

File: sphinx/source/docs/user_guide/geo.rst (hunk 1 above)
Comment (tcmetzger) [reply sphinx/source/docs/user_guide/geo.rst:7]: This page will still need some sort of intro, though. How about something short along the lines of:

> Bokeh supports creating map-based visualizations and working with geographical data.

File: sphinx/source/docs/user_guide/geo.rst [hunk 2]
```diff
@@ -1,59 +1,54 @@
...
 
 .. _userguide_geo_tile_provider_maps:
 
-Tile Provider Maps
+Tile provider maps
 ------------------
 
-Bokeh plots can also consume XYZ tile services which use the Web Mercator projection.
+Bokeh plots can consume XYZ tile services which use the Web Mercator projection.
```
Comment (tcmetzger) [inline sphinx/source/docs/user_guide/geo.rst:14]: @bryevdv Would we be clearer by saying something like: 

> Bokeh is compatible with several XYZ tile services that use the Web Mercator projection.

File: sphinx/source/docs/user_guide/geo.rst [hunk 3]
```diff
@@ -1,59 +1,54 @@
...
 The module :ref:`bokeh.tile_providers` contains several pre-configured tile sources with
-appropriate attribution which can be added to a plot using the
-:func:`~bokeh.models.plots.Plot.add_tile` method.
+appropriate attribution. To add these to a plot use the method
```
Comment (tcmetzger) [inline sphinx/source/docs/user_guide/geo.rst:16]: I believe there should be a comma after "plot".

>  To add these to a plot, use the method

File: sphinx/source/docs/user_guide/geo.rst [hunk 4]
```diff
@@ -1,59 +1,54 @@
...
 .. _userguide_geo_google_maps:
 
 Google Maps
 -----------
 
-Bokeh can also plot glyphs over a Google Map using the :func:`~bokeh.plotting.gmap`
-function. You must pass a `Google API Key`_ to this function in order for it to work, as
-well as any :class:`~bokeh.models.map_plots.GMapOptions` to configure the Google Map
-underlay. The Google API Key will be stored in the Bokeh Document JSON.
+To plot glyphs over a Google Map use the function :func:`~bokeh.plotting.gmap`.
```
Comment (tcmetzger) [inline sphinx/source/docs/user_guide/geo.rst:31]: There should be a comma after "Map"

File: sphinx/source/docs/user_guide/geo.rst [hunk 5]
```diff
@@ -1,59 +1,54 @@
...
+For the function to work you must pass it a `Google API Key`_ and configure the Google Map underlay :class:`~bokeh.models.map_plots.GMapOptions`.
```
Comment (tcmetzger) [inline sphinx/source/docs/user_guide/geo.rst:32]: Add a comma after "work".

**Review by tcmetzger — COMMENTED** [review]
Thank you very much for your edits!

**Comment by pillemer** [comment]
Thanks for the opportunity! I've made the corrections you pointed out.  Let me know if anything else needs to be done.

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thank you @pillemer !

**Pull request merged**